components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/RoutingObjectHandlerTest.kt

```diff
@@ -0,0 +1,274 @@
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.hotels.styx.Environment
+import com.hotels.styx.api.Eventual
+import com.hotels.styx.api.HttpRequest
+import com.hotels.styx.api.HttpResponseStatus.CREATED
+import com.hotels.styx.api.HttpResponseStatus.NOT_FOUND
+import com.hotels.styx.api.HttpResponseStatus.OK
+import com.hotels.styx.routing.RoutingObjectRecord
+import com.hotels.styx.routing.config.RoutingObjectFactory
+import com.hotels.styx.routing.db.StyxObjectStore
+import com.hotels.styx.routing.handlers.StaticResponseHandler
+import com.hotels.styx.server.HttpInterceptorContext
+import io.kotlintest.matchers.types.shouldBeTypeOf
+import io.kotlintest.matchers.types.shouldNotBeSameInstanceAs
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.FeatureSpec
+import io.mockk.mockk
+import reactor.core.publisher.Mono
+import java.nio.charset.StandardCharsets.UTF_8
+
+class RoutingObjectHandlerTest : FeatureSpec({
+
+    val environment = Environment.Builder().build()
+
+    val routeDatabase = StyxObjectStore<RoutingObjectRecord>()
+
+    val objectFactory = RoutingObjectFactory(environment, routeDatabase, listOf(), mockk(), true)
+
+    feature("Route database management") {
+        scenario("Injecting new objects") {
+            val handler = RoutingObjectHandler(routeDatabase, objectFactory)
+
+            handler.handle(
+                    HttpRequest.put("/admin/routing/objects/staticResponse")
+                            .body("""
+                            name: "staticResponse"
+                            type: "StaticResponseHandler"
+                            tags: []
+                            config:
+                               status: 200
+                               content: "Hello, world!"
+                            """.trimIndent(),
+                                    UTF_8)
+                            .build()
+                            .stream(), HttpInterceptorContext.create())
+                    .flatMap { it.aggregate(2000) }
+                    .toMono()
+                    .block()
+                    ?.status() shouldBe CREATED
+
+            routeDatabase.get("staticResponse").isPresent shouldBe true
+            routeDatabase.get("staticResponse").get().type shouldBe "StaticResponseHandler"
+            routeDatabase.get("staticResponse").get().config.shouldBeTypeOf<ObjectNode>()
+            routeDatabase.get("staticResponse").get().handler.shouldBeTypeOf<StaticResponseHandler>()
+        }
+
+        scenario("Retrieving objects") {
+            val handler = RoutingObjectHandler(routeDatabase, objectFactory)
+
+            handler.handle(
+                    HttpRequest.put("/admin/routing/objects/staticResponse")
```

Comment:
There are lots of tests that are set up this way. There should be some kind of shared setUp method for them (not for the test name `Injecting new objects` though, since the `put` is the thing being tested there).